lib/browserslist_useragent/resolver.rb

```diff
@@ -5,14 +5,16 @@
 module BrowserslistUseragent
   # Resolves uaser agent string to {family, version} hash
   class Resolver
+    USER_AGENT_PARSER = UserAgentParser::Parser.new
+    
     attr_reader :user_agent_string
 
     def initialize(user_agent_string)
       @user_agent_string = user_agent_string
     end
 
     def call
-      agent = UserAgentParser.parse(user_agent_string)
+      agent = USER_AGENT_PARSER.parse(user_agent_string)
```

Comment:
What do you think if we will define the singleton method with memoization?
```ruby
def self.parser
  @parser ||= UserAgentParser::Parser.new
end
```

It will be better for the application loading performance> Constants are initialized immediately when class code is evalutated, and it could cause application slowing down at start time

Reply:
Sure, that makes perfect sense.